src/Libraries/DSCPython/CPythonEvaluator.cs

```diff
@@ -447,6 +447,51 @@ private static string GetTraceBack(Exception e)
 
         #region Marshalling
 
+        /// <summary>
+        /// Add additional data marshalers to handle host data.
+        /// </summary>
+        [SupressImportIntoVM]
+        public override void RegisterHostDataMarshalers(object dataMarshaler)
+        {
+            DataMarshaler dataMarshalerToUse = dataMarshaler as DataMarshaler;
+            dataMarshalerToUse?.RegisterMarshaler((PyObject pyObj) =>
+            {
+                IntPtr gs = PythonEngine.AcquireLock();
+                try
+                {
+                    using (Py.GIL())
```

Comment:
I think Py.GIL() already sets up a Lock (like in line 459)

Reply:
+1 you should not need to set up your own lock.

Reply:
Yeah, ok, that's how it was in D4R. And actually we don't need (Py.GIL()) either since we are single threaded but I'll leave it as is for now. I see we are using both Acquire and Py.GIL() in several places - there will be another PR with optimizations.